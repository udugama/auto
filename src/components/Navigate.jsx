import React from 'react';
import { NavigateNext, NavigateBefore } from '@material-ui/icons';
import { Button } from '@material-ui/core';

export default class Navigate extends React.Component {

    render() {
        return (
            <Button
                variant="contained"
                color="default"
                onClick={this.props.onClick}
                startIcon={(this.props.name === 'Next') ? (<NavigateNext />) : (<NavigateBefore />)}
            >
                {this.props.name}
            </Button>
        );
    }    
}