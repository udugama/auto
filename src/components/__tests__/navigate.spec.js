import React from 'react';
import Navigate from './../Navigate';
import { shallow } from 'enzyme';
import { render } from '@testing-library/react';

it('renders without crashing', () => {
    shallow(<Navigate name='next' />);
});

it('render navigate button ', () => {
  const { getByText } = render(<Navigate name='next' />);
  expect(getByText).toMatchSnapshot();
});