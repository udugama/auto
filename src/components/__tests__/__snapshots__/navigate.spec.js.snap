// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render navigate button  1`] = `[Function]`;
